components/director/internal/domain/formation/service.go

```diff
@@ -469,6 +484,12 @@ func (s *service) AssignFormation(ctx context.Context, tnt, objectID string, obj
 			return nil, err
 		}
 
+		// When it is in initial state, the notification generation will be handled by the async API via resynchronizing the formation later
+		if formationFromDB.State != model.ReadyFormationState {
```

Comment:
In the case of deleting or delete error, we probably shouldn't assign to the formation at all. But in the case of initial or create_error, it should be added to the formation as label definition and formation assignments that stay in initial states without sending notifications. If the status api reports success, then it will send the notifications. If it is stuck in create error and somebody resynchronized the formation and it succeeds, then the formation lifecycle notification should call resynchronize and send the rest of the notifications for everything that has been added to the formation before it was "ready".

Reply:
refs - https://github.com/kyma-incubator/compass/pull/2928#discussion_r1130832270